content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md

```diff
@@ -0,0 +1,320 @@
+    SkyWalking, service mesh, istio, application performance monitoring
+---
+## Introduction
+
+In [the first blog](https://developer.hpe.com/blog/how-to-deploy-application-across-hybrid-clouds-beyond-hpe-greenlake-for-private-cloud-enterprise/), we described the detailed process of deploying the complex Online Boutique application to a hybrid cloud environment, across the public EKS cluster from AWS to the private Kubernetes cluster in [HPE GreenLake for Private Cloud Enterprise](https://www.hpe.com/us/en/greenlake/private-cloud-enterprise.html). This hybrid cloud model amplifies the benefits of both private and public clouds and allows for more seamless integration across technological barriers. It enables the enterprise to rely on the security of on-premises data centers while turning to the agility of managing the frontend of an application in public cloud. This model is now becoming increasingly popular as more businesses and enterprises shift toward cloud-based computing. However, this evolution presents challenges for monitoring, as applications and services are inherently more distributed in this environment. Having a good application performance monitoring (APM) tool is becoming essential within the hybrid cloud environment. It can consolidate performance metrics and troubleshooting data for assets across the hybrid cloud environment into a single application. It makes it easier to track metrics and allows enterprise to resolve problems quickly.
+
+In this blog post, we c﻿hoose the Apache SkyWalking as an APM tool and describe how to set up it, as a *self-hosted* APM tool, for monitoring and alerting the application performance across hybrid cloud environment. We take the service mesh as an auto instrumentation mechanism for application monitoring, without adding any manual process to instrument existing applications. 
+
+## Apache SkyWalking
+
+[Apache SkyWalking](https://skywalking.apache.org/) is an open source APM tool with capabilities for monitoring, tracing and diagnosing distributed system. It’s especially designed for microservices, cloud native and container-based architectures. 
+
+Apart from a list of supported agents to be used for instrumenting applications, Apache SkyWalking implements an [Envoy Access Log Service (ALS)](https://www.envoyproxy.io/docs/envoy/v1.18.2/api-v2/service/accesslog/v2/als.proto) based solution to provide observability on the service mesh, no matter the architecture or language. Since service mesh provides full control of *RPC*, this observation solution is much easier to be added without language specific technology. With this solution, users could get the application service topology map, metrics graph, request details and error message with a very nice visualization. This integration solution can be extremely important for monitoring and visualizing application that consists of many microservices running across on-premises, cloud-based or hybrid environments. 
```

Comment:
Since service mesh provides full control of ***RPC** communication protocol, this observation solution...